mountinfo/mountinfo_linux.go

```diff
@@ -105,16 +105,9 @@ func GetMountsFromReader(r io.Reader, filter FilterFunc) ([]*Info, error) {
 		p.Options = fields[5]
 
 		// zero or more optional fields
-		switch {
-		case sepIdx == 6:
-			// zero, do nothing
-		case sepIdx == 7:
-			p.Optional = fields[6]
-		default:
-			p.Optional = strings.Join(fields[6:sepIdx-1], " ")
```

Comment:
Looks like there's an off-by-one bug here; removing this switch caused it to panic;

```
--- FAIL: TestMountedBy (0.01s)
panic: runtime error: slice bounds out of range [6:5] [recovered]
	panic: runtime error: slice bounds out of range [6:5]
```

- we return an error if `sepIdx == 5` (earlier check)
- for `sepIdx == 6` we skipped (in this switch)
- for `7`, we take field 6 (equivalent to `fields[6:7]`, or `fields[6:sepIdx]`)
- for the "default" (else), we use `fields[6:sepIdx-1]`, which for (e.g.) `sepIndex=8` would do the same as `sepIdx=7`, so we'd drop one value